### PR TITLE
Fix for testagent restart issue

### PR DESCRIPTION
**Pull request by AshwiniChalla**
Based on the version avoid installing test agent when not required

### CODE REVIEW

**Comment by AshwiniChalla** [comment]
related bug : Bug 320689:even if agent version is same what is installed on machine, it installs and restarts machine.

**Comment by acesiddhu** [comment]
this has to go in releases/m83

File: Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1 [hunk 1]
```diff
@@ -3,13 +3,24 @@ function Install-Product($SetupPath, $UserName, $Password, $ProductVersion, $Arg
...
 	$InstalledCheckRegValueData = "1"
-
-	$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName -InstalledCheckRegValueData $InstalledCheckRegValueData
-        $testAgentFileExists = Test-Path "$env:SystemDrive\TestAgent\testagent"
-
-	if($testAgentFileExists -and $isProductExists)
+                
+	$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName         
+
+        $versionToInstall = ((get-item $SetupPath).VersionInfo.FileVersion)
```
Comment (acesiddhu) [inline Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1]: Get-Item

**Comment by AshwiniChalla** [comment]
Testing done: Verified the working of all possible scenarios. (fresh installation, existing lower equal and higher versions)

File: Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1 [hunk 2]
```diff
@@ -3,13 +3,19 @@ function Install-Product($SetupPath, $UserName, $Password, $ProductVersion, $Arg
...
-	if($testAgentFileExists -and $isProductExists)
+                
+	$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName         
+
+        $versionToInstall = ((Get-Item $SetupPath).VersionInfo.FileVersion) 
+        $versionInstalled = (Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName "version")
+
+        if($versionToInstall -ne $null)
+        {
+		$versionToInstall = $versionToInstall.SubString(0, $versionToInstall.LastIndexOf('.'))
```
Comment (acesiddhu) [inline Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1]: indentation

**Pull request merged**